core/src/test/java/kafka/test/junit/ClusterTestExtensions.java

```diff
@@ -140,8 +141,38 @@ private void generateClusterConfigurations(ExtensionContext context, String gene
         ReflectionUtils.invokeMethod(method, testInstance, generator);
     }
 
-    private void processClusterTest(ExtensionContext context, ClusterTest annot, ClusterTestDefaults defaults,
-                                    Consumer<TestTemplateInvocationContext> testInvocations) {
+    private List<TestTemplateInvocationContext> processClusterTests(ExtensionContext context,
+                                                                   ClusterTestDefaults defaults) {
+
+        ClusterTests clusterTestsAnnot = context.getRequiredTestMethod().getDeclaredAnnotation(ClusterTests.class);
+        List<TestTemplateInvocationContext> ret = new ArrayList<>();
+
+        if (clusterTestsAnnot != null) {
+            for (ClusterTest annot : clusterTestsAnnot.value()) {
+                ret.addAll(processClusterTestInternal(context, annot, defaults));
+            }
+        }
+
+        if (ret.isEmpty()) {
+            throw new IllegalStateException("processClusterTests method should provide at least one config");
+        }
+
+        return ret;
+    }
+
+    private List<TestTemplateInvocationContext> processClusterTest(ExtensionContext context, ClusterTest annot,
+                                                                   ClusterTestDefaults defaults) {
+        List<TestTemplateInvocationContext> ret = processClusterTestInternal(context, annot, defaults);
+
+        if (ret.isEmpty()) {
+            throw new IllegalStateException("processClusterTest method should provide at least one config");
+        }
+
+        return ret;
+    }
+
+    private List<TestTemplateInvocationContext> processClusterTestInternal(ExtensionContext context, ClusterTest annot,
```

Comment:
It seems we don't need `processClusterTestInternal` as it just has a check. We can move the check to `processClusterTest`

Reply:
`processClusterTestInternal ` is used by `processClusterTest` and `processClusterTests` so we need to keep it.

Reply:
Sorry for my unclear comments.

My point was they can be merged into single method.